crates/sui-types/src/messages.rs

```diff
@@ -343,6 +343,21 @@ impl SingleTransactionKind {
         self.shared_input_objects().next().is_some()
     }
 
+    /// Returns a tuple of (# mutable, # immutable) shared input objects
+    /// Currently used to charge for the RW Locks
+    pub fn shared_input_objects_counts(&self) -> (usize, usize) {
+        let (mut mutable_count, mut immutable_count) = (0, 0);
+
+        for s in self.shared_input_objects() {
+            if s.mutable {
+                mutable_count += 1;
+            } else {
+                immutable_count += 1;
+            }
+        }
+        (mutable_count, immutable_count)
```

Comment:
Could you rewrite this as something like:

```rust

let (mutable, immutable) = self.shared_input_objects().iter().partition(|s| s.mutable);
(mutable.len(), immutable.len())

```

IMO this would feel like a more "Rusty" way to write it. But opinions may vary on this..

Reply:
You're correct! Thanks for this handy feature.

Reply:
Okay technically isn't this more expensive? I'm creating two intermediate containers which I actually don't need.